File: Practicas/EjerLibroPascual/intercalar.pl
%%% INTERCALAR
% intercalar(L1, L2, L): entrelaza las listas L1 y L2 para originar una lista L.
intercalar(L1, L2, L) :- intercalar(L1, L2, _, L).
intercalar([], [], [], []).
intercalar([X|L1], [Y|L2], [X,Y|I], L) :- intercalar(L1, L2, I, L).

File: Haskell/intercalar.hs
-- Intercalar --

-- Entrelaza dos listas para obtener una tercera --

intercalar :: [a] -> [a] -> [a]

intercalar [] [] = []
intercalar [] ys = ys
intercalar xs [] = xs
intercalar (x:xs) (y:ys) = x:y:(intercalar xs ys)